figures/mapping-histology-labels.Rmd

```diff
@@ -56,284 +46,340 @@ Rscript -e "rmarkdown::render('figures/mapping-histology-labels.Rmd',
+                   "Ganglioglioma",
+                   "Low-grade glioma astrocytoma",
+                   "Meningioma",
+                   "Ewing sarcoma",
+                   "Dysembryoplastic neuroepithelial tumor",
+                   "Neurofibroma Plexiform",
+                   "Schwannoma",
+                   "Craniopharyngioma"),
+  cancer_group_hex = c("#4d2635",
+                       "#bf0099",
+                       "#ff40d9",
+                       "#ffccf5",
+                       "#4d0d85",
+                       "#b08ccf",
+                       "#a340ff",
+                       "#2200ff",
+                       "#058aff",
+                       "#8c8cff",
+                       "#000080",
+                       "#2db398",
+                       "#9fbf60",
+                       "#614e01",
+                       "#e6ac39",
+                       "#ab7200",
+                       "#b33000"),
+  stringsAsFactors = FALSE
+)
 ```
 
-Make this notebook stop if there are more than 16 histology groups + `Normal`. 
+And again, we'll create a legend with `legend()`
 
 ```{r}
-if (nrow(display_group_df) > 18) {
-  stop("There are more than 18 categories in `display_group`. We may want to re-evaluate the high-level histology groupings")
-}
+plot(NULL, xaxt = "n", yaxt = "n", bty = "n", ylab = "", xlab = "", 
+     xlim = 0:1, ylim = 0:1)
+legend("topleft", 
+       legend = cancer_group_df$cancer_group,
+       col = cancer_group_df$cancer_group_hex,
+       pch = 15, pt.cex = 1.5, cex = 0.75, bty = "n")
+mtext("Cancer Group", at = 0.0625, cex = 1)
 ```
 
-# Make `display_order`
+### Output
 
-Get ranks in order of big to small and make them into a new column in `display_group_df`. 
-We will always want the "normal", "benign", "other_tumor" groups to come last so we will push then to the end of the factor order. 
+We can create a data frame that contains both palettes with a series of left joins, where we will then fill the NA values with a single (gray) hex code per column (`#808080` for `broad_histology`, `#d3d3d3` for `cancer_group`.)
 
 ```{r}
-display_order_df <- display_group_df %>% 
-  dplyr::mutate(display_group = forcats::fct_reorder(display_group, n, .desc = TRUE) %>%
-                  forcats::fct_relevel("benign", "other tumor", "normal", after = Inf),
-                display_order = as.numeric(display_group)) # save the factor order for text table export
+palette_df <- histologies_df %>%
+  # Exclude normal samples
+  filter(sample_type == "Tumor") %>%
+  # Filter to unique broad histology--cancer group pairs
+  select(broad_histology, 
+         cancer_group) %>% 
+  distinct() %>%
+  # Add broad histology palette
+  left_join(broad_histology_df, by = "broad_histology") %>%
+  # Add cancer group palette
+  left_join(cancer_group_df, by = "cancer_group") %>%
+  # Fill all other values with gray colors
+  replace_na(list(broad_histology_hex = "#808080", 
+                  cancer_group_hex = "#b5b5b5")) %>%
```

Comment:
Because we use these twice, maybe we should pull them out to variables (defined earlier, with the data frames, presumably).
```suggestion
  replace_na(list(broad_histology_hex = broad_histology_other_hex,
                  cancer_group_hex = cancer_group_other_hex)) %>%
```